resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -0,0 +1,307 @@
+if ([string]::IsNullOrEmpty($env:TestRegistryPath)) {
+    $global:tzAutoUpdatePath = 'HKLM:\SYSTEM\CurrentControlSet\Services\W32Time\Parameters'
+    $global:timeZoneInformationPath = 'HKLM:\SYSTEM\CurrentControlSet\Control\TimeZoneInformation'
+} else {
+    $global:tzAutoUpdatePath = $global:timeZoneInformationPath = $env:TestRegistryPath
+}
+
+#region Functions
+function TryGetRegistryValue {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$Key,
+
+        [Parameter(Mandatory = $true)]
+        [string]$Property
+    )
+
+    if (Test-Path -Path $Key) {
+        try {
+            return (Get-ItemProperty -Path $Key -Name $Property -ErrorAction SilentlyContinue | Select-Object -ExpandProperty $Property)
+        } catch {
+            Write-Verbose "Property `"$($Property)`" could not be found."
+        }
+    } else {
+        Write-Verbose 'Registry key does not exist.'
+    }
+}
+function Get-ValidTimeZone {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $TimeZone
+    )
+
+    try {
+        $timeZoneId = (Get-TimeZone -Id $TimeZone -ErrorAction Stop).Id
+    } catch {
+        throw [System.Configuration.ConfigurationException]::new("Executing 'Get-TimeZone' failed. Error: $($PSItem.Exception.Message)")
+    }
+
+    return $timeZoneId
+}
+#endRegion Functions
+
+#region Classes
+<#
+.SYNOPSIS
+    This `Time` DSC Resource allows you to manage the time zone, automatic time zone update, and system tray date/time visibility settings on a Windows machine.
+
+.DESCRIPTION
+    This `Time` DSC Resource allows you to manage the time zone, automatic time zone update, and system tray date/time visibility settings on a Windows machine.
+
+.PARAMETER TimeZone
+    The time zone to set on the machine. The value should be a valid time zone ID from the list of time zones (Get-TimeZone -ListAvailable).Id. The default value is the current time zone.
+
+.PARAMETER SetTimeZoneAutomatically
+    Whether to set the time zone automatically. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Set time automatically.
+
+.PARAMETER AdjustForDaylightSaving
+    Whether to adjust for daylight saving time. The value should be a boolean. You can find the setting in `Settings -> Time & Language -> Date & Time -> Adjust for daylight saving time automatically.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -Name Time -Method Set -Property @{ TimeZone = "Pacific Standard Time"}
+
+    This example sets the time zone to Pacific Standard Time.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -Name Time -Method Get -Property {}
+
+    This example gets the current time settings on the machine.
+#>
+[DscResource()]
+class TimeZone {
+    [DscProperty(Key)]
+    [string] $TimeZone
```

Comment:
nit: I feel that this should be `TimeZoneId` or `Id` because that is what I see from the output of `Get-TimeZone`

Reply:
Changed.